Fix compiler warings, mainly unused variables

fjansson opened the pull request:
372 warnings less.
* changed line endings of modtracers DOS->Unix
* did not touch modbulkmicroc sine it's being worked on
* did not touch RRTMG*

fjansson committed:
Fix compiler warings, mainly unused variables

372 warnings less.
* changed line endings of modtracers DOS->Unix
* did not touch modbulkmicroc sine it's being worked on
* did not touch RRTMG*

CasparJungbacker reviewed (approved):
Nice, the compilation output is noticeably cleaner. A lot of the left-over warnings seem to come from variables related to time-averaging (real to integer), so I'm wondering what the best way to do this explicitly is.

fjansson committed:
Fix compiler warnings, add explicit rounding to integers

idtav = dtav/tres        -> idtav = int(dtav / tres, kind=kind(idtav))
itimeav = timeav/tres    -> itimeav = int(timeav / tres, kind=kind(itimeav))
nsamples = itimeav/idtav -> nsamples = int(itimeav / idtav)

removes 52 warnings like "Possible change of value in conversion from REAL(8) to INTEGER(8)"

fjansson committed:
shr_orb_mod: fix compiler warnings about real precision

adding explicit _SHR_KIND_R4 to numerical constants
or changing real numerical constants to integers (1.0 to 1) when safe.
457 warnings less.

fjansson committed:
Compiler warnings: adjust gfortran compiler flags

Add -Wno-tabs -Wno-compare-reals after -Wall for gfortran.

-130 warnings for compare-reals.
Our real comparisons almost always check for == 0 or == -1
these are well defined.

-837 for tabs.
basically all of them in one of the RRTMG modules which we
don't want to mess with at this point

fjansson committed:
modcufft: suppress warning about unused arguments in empty functions

fjansson committed:
Fix compiler warnings

missing & in multiline strings
some unused variables
dos2unix on moddrydeposition and moddepcrossection
